samples/GstAudioVideoReceiver.c

```diff
@@ -0,0 +1,230 @@
+#include "Samples.h"
+#include <gst/gst.h>
+#include <gst/app/app.h>
+#include <gst/app/gstappsink.h>
+
+// This function is a callback for the transceiver for every single video frame it receives
+// It writes these frames to a buffer and pushes it to the `appsrcVideo` element of the
+// GStreamer pipeline created in `receiveGstreamerAudioVideo`. Any logic to modify / discard the frames would go here
+VOID onGstVideoFrameReady(UINT64 customData, PFrame pFrame)
+{
+    GstFlowReturn ret;
+    GstBuffer* buffer;
+    GstElement* appsrcVideo = (GstElement*) customData;
+    if (!appsrcVideo) {
+        DLOGE("Null");
+        return;
+    }
+    if (pFrame == NULL) {
+        DLOGE("Null frame received!, skipping it");
+        return;
+    }
+    buffer = gst_buffer_new_allocate(NULL, pFrame->size, NULL);
+    if (!buffer) {
+        DLOGE("Buffer allocation failed");
+        return;
+    }
+
+    DLOGD("Frame size: %d, presentationTs: %llu", pFrame->size, pFrame->presentationTs);
+    GST_BUFFER_PTS(buffer) = pFrame->presentationTs;
+    GST_BUFFER_DURATION(buffer) = gst_util_uint64_scale(1, GST_SECOND, DEFAULT_FPS_VALUE);
+    if (gst_buffer_fill(buffer, 0, pFrame->frameData, pFrame->size) != pFrame->size) {
+        DLOGE("Buffer fill did not complete correctly");
+        gst_buffer_unref(buffer);
+        return;
+    }
+    g_signal_emit_by_name(appsrcVideo, "push-buffer", buffer, &ret);
+    if (ret != GST_FLOW_OK) {
+        DLOGE("Error pushing buffer: %s", gst_flow_get_name(ret));
+    }
+    gst_buffer_unref(buffer);
+}
+
+// This function is a callback for the transceiver for every single audio frame it receives
+// It writes these frames to a buffer and pushes it to the `appsrcAudio` element of the
+// GStreamer pipeline created in `receiveGstreamerAudioVideo`. Any logic to modify / discard the frames would go here
+VOID onGstAudioFrameReady(UINT64 customData, PFrame pFrame)
+{
+    GstFlowReturn ret;
+    GstBuffer* buffer;
+    GstElement* appsrcAudio = (GstElement*) customData;
+    if (!appsrcAudio) {
+        DLOGE("Null");
```

Comment:
Can we make this a little more descriptive?

Reply:
Done